compose/src/main/java/com/stripe/android/compose/elements/SimpleOutlineFieldComposable.kt

```diff
@@ -0,0 +1,358 @@
+        bottomEnd = ZeroCornerSize,
+        bottomStart = ZeroCornerSize
+    )
+) {
+    // AnimatedVisibility(visible = autoCompleteState.isSearching) {
+    Log.d("Construct", "OptionalBorderTextField $debugLabel")
+    if (isBorder) {
+        CustomOutlinedTextField(
+            value = value,
+            modifier = modifier,
+            onValueChange = onValueChange,
+            label = { TextFieldLabel(label) },
+            isError = isError,
+            enabled = enabled,
+            readOnly = readOnly,
+            textStyle = textStyle,
+            placeholder = placeholder,
+            leadingIcon = leadingIcon,
+            trailingIcon = trailingIcon,
+            visualTransformation = visualTransformation,
+            keyboardOptions = keyboardOptions,
+            keyboardActions = keyboardActions,
+            singleLine = singleLine,
+            maxLines = maxLines,
+            interactionSource = interactionSource,
+        )
+    } else {
+        CustomTextFieldNoOutline(
+            debugLabel = debugLabel,
+            value = value,
+            modifier = modifier,
+            onValueChange = onValueChange,
+            label = { TextFieldLabel(label) },
+            isError = isError,
+            enabled = enabled,
+            readOnly = readOnly,
+            textStyle = textStyle,
+            placeholder = placeholder,
+            leadingIcon = leadingIcon,
+            trailingIcon = trailingIcon,
+            visualTransformation = visualTransformation,
+            keyboardOptions = keyboardOptions,
+            keyboardActions = keyboardActions,
+            singleLine = singleLine,
+            maxLines = maxLines,
+            interactionSource = interactionSource,
+            shape = shape,
+        )
+    }
+
+}
+
+@Composable
+fun CustomOutlinedTextField(
+    debugLabel: String = "unknown",
+    value: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    readOnly: Boolean = false,
+    textStyle: TextStyle = LocalTextStyle.current,
+    label: @Composable (() -> Unit)? = null,
+    placeholder: @Composable (() -> Unit)? = null,
+    leadingIcon: @Composable (() -> Unit)? = null,
+    trailingIcon: @Composable (() -> Unit)? = null,
+    isError: Boolean = false,
+    visualTransformation: VisualTransformation = VisualTransformation.None,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    keyboardActions: KeyboardActions = KeyboardActions(),
+    singleLine: Boolean = false,
+    maxLines: Int = Int.MAX_VALUE,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+    Log.d("Construct", "CustomOutlinedTextField $debugLabel")
+    val colors = TextFieldDefaults.outlinedTextFieldColors(
+        textColor = if (isError) {
+            MaterialTheme.colors.error
+        } else {
+            LocalContentColor.current.copy(LocalContentAlpha.current)
+        }
+    )
```

Comment:
Could we combine the checks (and variations) for `isBorder` and `colors` into a single element? Even though it would increase the local complexity of the function, it would help lower the overall complexity of handling all these variants. The higher up we handle these checks the better since there are only 2 possible options: `TextField` and `OutlinedTextField`, the colour is applicable to both of these widgets.

Reply:
I will take a look at better ways of simplifying this.   It might actually make sense for now just to eliminate the border until it is really a style we want to use.